model/diagnostic/Paginator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ *
+ */
+
+namespace oat\taoClientDiagnostic\model\diagnostic;
+
+use oat\taoClientDiagnostic\model\storage\PaginatedStorage;
+
+/**
+ * Provides common data helper for datatable component.
+ */
+class Paginator
```

Comment:
yes, JS is refactoring the proctoring, all duplicate code will be merge in tao, close to datatable.
This part will be merge in tao